### DISCUSSION TITLE
Get Dynamic Block Properties by ezdxf

### DISCUSSION THREAD

**Discussion by twnhushlc** [mozman/ezdxf#403] [Q&A] [answered]
Hi, 
I am a Taiwanese student, sorry about the Chinese GUI in the pictures

I create a block which have Linear Parameters to measure my block size and show it on "Custom" in the "PROPERTIES",
Like this web shows, https://grabcad.com/tutorials/dynamic-block-in-autocad
So Far, I can use `insert.attribs` to get my block's attributes, but I also need a way to get Dynamic Block Properties 
I am wondering, is there a function or method to get Dynamic Block Properties in ezdxf
I have tried many method like `BlockLayout` `Block` `Insert` `Attrib` `AttDef`, but none of these get the value I need.

Which method I need to use to get 'LeftUpX', 'LeftDownY' values ?

![BLOCK](https://user-images.githubusercontent.com/82866873/115379033-b9c18080-a203-11eb-9271-8066562b2a3b.png)
![PARAMETER](https://user-images.githubusercontent.com/82866873/115379037-bb8b4400-a203-11eb-9ad9-6edc5e9292d5.png)
![PROPERTIES](https://user-images.githubusercontent.com/82866873/115379039-bc23da80-a203-11eb-881d-d2aabdee3343.png)

Thanks,
Pete

**Comment by mozman** [comment, marked as answer]
The structure of dynamic blocks is not documented. Even BricsCAD can not create dynamic blocks and therefore exist no active support for these feature by ezdxf.

**Comment by twnhushlc** [comment]
sir, 
I do some research about AutoCAD's ActiveX, there is a way to get dynamic blocks properties. Is there possible that ezdxf will develop functions to support this ?
![TREE](https://user-images.githubusercontent.com/82866873/115645138-69a50400-a352-11eb-93f0-42183c3f23a3.png)

Thanks, 
Pete

**Comment by mozman** [comment]
No, that is not the way ezdxf works, ezdxf is a straight file format interface that read and write the content of DXF files. If ezdxf would utilizes external DLLs to import and export arbitrary CAD content the package would depend on third party software, in this case on the installation of AutoCAD, and it would only work on Windows (maybe macOS).
